homeassistant/components/automation/__init__.py

```diff
@@ -313,16 +318,37 @@ def as_dict(self) -> Dict[str, Any]:
         return result
 
 
+class LimitedSizeDict(OrderedDict):
```

Comment:
All dictionaries are ordered in Python land nowadays. You could probably just inherit `dict` here.

Reply:
We can't extend `dict` because it doesn't allow popping the first added item, only the last added.